webapp/src/main/java/com/box/l10n/mojito/pseudoloc/PseudoLocalization.java

```diff
@@ -0,0 +1,203 @@
+		pseudoLocMap.put("v", "ν");
+		pseudoLocMap.put("w", "ŵωώшщẁẃẅ");
+		pseudoLocMap.put("W", "ŴШЩẀẂẄ");
+		pseudoLocMap.put("x", "×хж");
+		pseudoLocMap.put("X", "ΧχХЖ");
+		pseudoLocMap.put("y", "ýÿŷγУўỳу");
+		pseudoLocMap.put("Y", "ÝŶΎΥΫỲЎ");
+		pseudoLocMap.put("z", "źżž");
+		pseudoLocMap.put("Z", "ŹŻΖŽ");
+	}
+
+	/**
+	 * Pseudo localize a string depending on instance settings to choose the type, expansion factor and bracket option.
+	 * If pseudo-localization is disabled, returns the passed in string
+	 *
+	 * @param string string to be pseudo localized
+	 * @return pseudo localized string
+	 */
+	public String convertStringToPseudoLoc(String string)
+	{
+		String expanded = "";
+
+		StringBuilder sb = new StringBuilder();
+
+		if (Strings.isNullOrEmpty(string)) {
+			expanded = convertAsciiToDiacritics(string);
+			expanded = expand(expanded);
+			sb.insert(0,'⟦');
+			sb.append(expanded);
+			sb.append('⟧');
+
+			expanded = sb.toString();
+		}
+
+		return expanded;
+	}
+
+	/**
+	 * Expands the given string by the expansion factor with some unicode characters.
+	 *
+	 * @param string
+	 * @return
+	 */
+	public String expand(String string) {
+
+		int expandedStringLength = (int) Math.ceil(string.length() * EXPANSION_FACTOR / 100.0);
+
+		StringBuilder str = new StringBuilder();
+
+		for (int i = 0; i < expandedStringLength; i++) {
+			if (i == (expandedStringLength - 1)) {
+				str.append(" ");
+			} else {
+				str.append(BEGINNING_CHAR);
+			}
+		}
+
+		str.append(string);
+
+		for (int i = 0; i < expandedStringLength; i++) {
+			if (i == 0) {
+				str.append(" ");
+			} else {
+				str.append(ENDING_CHAR);
+			}
+		}
+
+		return str.toString();
+	}
+
+	/**
+	 * Converts ASCII letter into equivalent characters with accent/diacritics
+	 *
+	 * @param string String to be converted
+	 * @return
+	 */
+	public String convertAsciiToDiacritics(String string) {
+		int stringLength = string.length();
+		StringBuilder str = new StringBuilder();
+		for (int i = 0; i < stringLength; i++) {
+			String character = substr(string, i, 1);
+			str.append(getMappingCharFromMap(character));
+		}
+
+		return str.toString();
+	}
+
+	/**
+	 * Get a non ASCII character mapping to provided character or the character itself if there is no mapping
+	 *
+	 * @param character ASCII character to be mapped
+	 * @return Non ASCII character or character itself
+	 */
+	private String getMappingCharFromMap(String character) {
+		String mappedChar = character;
+
+		String mappingCharsForChar = pseudoLocMap.get(mappedChar);
+
+		if (mappingCharsForChar != null) {
+
+			int matchesLength = mappingCharsForChar.length();
+			int min = 0 + (int) (Math.random() * (matchesLength - 1));
+			mappedChar = substr(mappingCharsForChar, min, 1);
+		}
+
+		return mappedChar;
+	}
+
+	/**
+	 * This is the equivalent to PHP substr method.
+	 * @param string
+	 * @param from
+	 * @param to
+	 * @return
+	 */
+	private String substr(String string, int from, int to) {
```

Comment:
I need to keep this function because what the native .substring does is not what i want done.